Framework/MDAlgorithms/src/IntegrateQLabEvents.cpp

```diff
@@ -301,6 +314,25 @@ PeakShapeEllipsoid_const_sptr IntegrateQLabEvents::ellipseIntegrateEvents(
   inti = peak_w_back.first - ratio * backgrd.first;
   sigi = sqrt(peak_w_back.second + ratio * ratio * backgrd.second);
 
+  if (inti < 0) {
+    std::ostringstream msg;
+    msg << "Negative intensity found: " << inti << "\n"
```

Comment:
You also pointed out the case when two peaks are very close which can happen when `findPeaks` reports an unreasonable number of peaks [like in this case](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/245#note_531913). For a given peak with associated negative intensity, would it be too difficult to check if its first neighbor peak lies withing the background shell?

Reply:
Based on my understanding, the peak information (qList) was converted to cell in the constructor of `IntegrateQLabEvents` before thrown away, which means we don't really have a way to query the neighboring information by the time we do the integration.
Alternatively, we could try at a higher level, `IntegrateEllipsoids`, but this would require us to build a KDTree based on peaks positions and probably a significant amount of changes to the current `IntegrateEllipsoids`.
I think the fundamental issue here is how we estimate the background, where the "chop off top 1%" method doesn't always work.
I believe we are better off developing a new way to estimate the background (such as selective median filter) so that the background estimate is not skewed by having neighboring peaks, and this kind of work is better off as a standalone user story.